### ISSUE TITLE
MQTT (tls test) hardware refused to connect.

### ISSUE THREAD

**Issue by FireWizard52** [domoticz/domoticz#5195]
@gizmocuz 

I upgraded two Raspberry Pi's to build 14284.

I saw you have been working on MQTT

According to Domoticz versions: Allow specifying server address with tls:// for secure connections, also port 8883 is considered a secure connection.

After upgrading both Raspberry Pi's, they refuse to connect with the MQTT servers

Message: Failed to start, return code: 14/Connection refused (Check IP/Port)

Does this require a configuration change? Either in Domoticz or at the MQTT server side?

Regards

**Comment by quim-net** [comment]
2022-03-16 22:19:41.943 Error: Local MQTT: Failed to start, return code: 14/Connection refused (Check IP/Port)

**Comment by gizmocuz** [comment]
No it does not and it is not ready yet.
But the idea is when you enter a address like tls://192.168.0.10 or use 192.168.0.10 with port 8883, and you have not specified a certificate, it should use the certificate from the startup folder.

`./server_cert.pem`

Maybe it can not find this file, you could try to enter a fixed location ?
As said, this is not finished yet

If that also does not work, do you have any messages in your MQTT broker log ?

Edit: Just tested on a different location, it connects by entering 8883 as port...
So check your broker log or try specifying the default certificate
But MQTT already worked securely when you specified a certificate before
Also check your TLS version, maybe you need to set this to 1.2

Reference code/commit:
https://github.com/domoticz/domoticz/commit/d25294bd64eef94d64566b3ddf43ae63abdc3366

**Comment by gizmocuz** [comment]
@quim-net, you are also trying to use tls or it does not connect in general ?

**Comment by gizmocuz** [comment]
It was caused by a typo in the code that has been corrected now, beta 14285 solves this
But if you could test with tls://<ip address> without specifying a certificate, or using normal address with port 8883,
to see if it connects correctly secure, that would be great
Of course you need to configure your broker to support secure connections

**Comment by quim-net** [comment]
good morning, problem solved with the new beta update 14285. thank you very much

**Comment by FireWizard52** [comment]
Hello @gizmocuz 

Indeed the problem is solved. I can connect with MQTT as before on port 1883.

I did also some other tests.

In the mosquitto.conf file and also in the conf.d directory in mymosquitto.conf I modified the port setting from 1883 to 8883 and restarted the mosquitto service.
In the Domoticz MQTT I modified the port setting as well to 8883.

It indicated the following:

` 2022-03-17 19:36:51.211 Status: MQTT RPi1: Connecting to 192.168.10.51:8883
2022-03-17 19:36:51.211 Status: MQTT RPi1: enabled TLS mode
`
I also used tls://192.168.10.51 and port 8883

`  2022-03-17 19:34:41.187 Status: MQTT RPi1: Connecting to tls://192.168.10.51:8883
2022-03-17 19:34:41.188 Status: MQTT RPi1: enabled TLS mode
`

Unfortunately it gives an error.

`2022-03-17 19:46:32.143 Error: MQTT RPi1: disconnected, restarting (rc=19/Unknown error.)`

The "server_cert.pem" is located in the /home/pi/domoticz directory.

I do use a username/password on my Mosquitto server, but that should not have any impact, I think.

Is there anything else I should do?
